src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -500,11 +500,12 @@ public void StopStake()
                         utxoStakeDescription.TxOut = utxo;
                         utxoStakeDescription.OutPoint = new OutPoint(set.TransactionId, infoTransaction.Transaction.Index);
                         utxoStakeDescription.Address = infoTransaction.Address;
-                        utxoStakeDescription.HashBlock = this.chain.GetBlock((int)set.Height).HashBlock;
+                        utxoStakeDescription.HashBlock = this.chain.GetBlock((int)set.Height)?.HashBlock;
```

Comment:
This is a good I would suggest to put the line`var blockHash = 
 this.chain.GetBlock((int)set.Height)?.HashBlock;` before the `if`
then use it as part of the condition 
` if ((utxo != null) && (utxo.Value > Money.Zero) && (blockHash != null))`

Reply:
yeah that would be even better

Reply:
Agree, it will be better.